core/chains/evm/txmgr/evm_tx_store_benchmark_test.go

```diff
@@ -0,0 +1,170 @@
+package txmgr_test
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	txmgrcommon "github.com/smartcontractkit/chainlink-framework/chains/txmgr"
+	txmgrtypes "github.com/smartcontractkit/chainlink-framework/chains/txmgr/types"
+	"github.com/smartcontractkit/chainlink-integrations/evm/assets"
+	"github.com/smartcontractkit/chainlink-integrations/evm/client/clienttest"
+	"github.com/smartcontractkit/chainlink-integrations/evm/testutils"
+	"github.com/smartcontractkit/chainlink-integrations/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txmgr"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/cltest"
+)
+
+func BenchmarkCreateTransactionTxStore(b *testing.B) {
+	db := testutils.NewSqlxDB(b)
+	txStore := newTxStore(b, db)
+	kst := cltest.NewKeyStore(b, db)
+
+	_, fromAddress := cltest.MustInsertRandomKey(b, kst.Eth())
+	toAddress := testutils.NewAddress()
+	gasLimit := uint64(1000)
+	payload := []byte{1, 2, 3}
+
+	ethClient := clienttest.NewClientWithDefaultChainID(b)
+
+	// queue under capacity inserts eth_tx
+	subject := uuid.New()
+	strategy := newMockTxStrategy(b)
+	strategy.On("Subject").Return(uuid.NullUUID{UUID: subject, Valid: true})
+
+	b.ResetTimer()
+	for txCount := 1; txCount <= b.N; txCount++ {
```

Comment:
This one seems like a good candidate for `b.Loop()` so you don't have to worry about off-by-one:
```suggestion
	for txCount := 1; b.Loop(); txCount++ {
```
Obviously it wasn't wrong now, but this way readers don't worry about double checking and future edits won't easily make a mistake.

Reply:
Ah nevermind. `b.Loop()` seems nuanced based on another comment. In that case, what about this?
```suggestion
	for n := 0; n < b.N; n++ {
		txCount := n+1
```